src/test/java/ch/ivyteam/ivy/maven/TestDeployToRunningEngine.java

```diff
@@ -97,7 +98,10 @@ public void canDeployRemoteIar() throws Exception
     Executor startedProcess = null;
     try
     {
+      System.setOut(originalOut);
       startedProcess = mojo.startEngine();
+      deployMojo.deployEngineUrl = (String)rule.project.getProperties().get(EngineControl.Property.TEST_ENGINE_URL);
+      System.setOut(new PrintStream(outContent));
```

Comment:
I see that you play with the logs ... but not any explicit assertions on the /context uri.
.... the best would be an explicit tests against 'evaluateDefaultContext' ... which would also drive the architecture (make static ... log in mojo)

Reply:
Another idea on logs: as the context evaluation is a thing that could fail in the future when the wrong engine version is combined with the plugin > we could write a debug log on the evaluated context dir. Which would help for failure analysis (mvn clean verify -X)
.... and maybe also use this log for test assertions?

Reply:
This change has nothing to do with the 'evaluateDefaultContext' function. It's only a fix for a test which was flaky. It needs a correct deployUrl because the default is not working in this test environment, and for this the startEngine method needs the log to evaluate the correct start Url. But later the log needs to be checked if the deployment worked so...

Why should the context evaluation fail in the future? Are there plans to change this url again? I've build it like this because now it is compatible with the old and the new behaviour. But i go with you some more debug logs would be nice for analysis!

I would like a test agains the 'evaluateDefaultContext' function, but the only thing that can be tested now I think, is that no error log occurs. Because if we test that 'ivy' needs to be returned this test fail as soon as the new behaviour is merged...